src/Nethermind/Nethermind.Optimism.Test/CL/DepositTransactionBuilderTest.cs

```diff
@@ -0,0 +1,723 @@
+// SPDX-FileCopyrightText: 2025 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using FluentAssertions;
+using Nethermind.Core;
+using Nethermind.Core.Extensions;
+using Nethermind.Core.Crypto;
+using Nethermind.Core.Test.Builders;
+using Nethermind.Facade.Eth;
+using Nethermind.Optimism.CL;
+using Nethermind.Optimism.CL.Derivation;
+using NUnit.Framework;
+using Nethermind.Int256;
+using Nethermind.JsonRpc.Data;
+
+namespace Nethermind.Optimism.Test.CL;
+
+[TestFixture]
+public class DepositTransactionBuilderTest
+{
+    private static readonly Hash256 SomeHash = new("0x73f947f215a884a09c953ffd171e3a3feab564dd67cfbcbd5ee321143a220533");
+    private static readonly Address DepositAddress = TestItem.AddressA;
+    private static readonly Address SomeAddressA = TestItem.AddressB;
+    private static readonly Address SomeAddressB = TestItem.AddressC;
+    private static readonly Address SomeAddressC = TestItem.AddressD;
+    private static readonly Address SomeAddressD = TestItem.AddressE;
+
+    private readonly CLChainSpecEngineParameters _engineParameters;
+    private readonly DepositTransactionBuilder _builder;
+
+    public DepositTransactionBuilderTest()
+    {
+        _engineParameters = new CLChainSpecEngineParameters { OptimismPortalProxy = DepositAddress };
+        _builder = new DepositTransactionBuilder(TestBlockchainIds.ChainId, _engineParameters);
+    }
+
+    [Test]
+    public void DeriveUserDeposits_NoDeposits()
+    {
+        ReceiptForRpc[] receipts = [];
+        var depositTransactions = _builder.BuildUserDepositTransactions(receipts).ToArray();
+
+        depositTransactions.ToArray().Length.Should().Be(0);
```

Comment:
Unnecessary `.ToArray()` in most tests:
```suggestion
        depositTransactions.Length.Should().Be(0);
```